File: Daily Test/Java/Free Time - No Appointments 9.1.2026.java
import java.util.*;

public class Hello {

    static int toMin(String t) {
        String[] p = t.split(":");
        return Integer.parseInt(p[0]) * 60 + Integer.parseInt(p[1]);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[][] arr = new int[n][2];

        for (int i = 0; i < n; i++) {
            arr[i][0] = toMin(sc.next());
            arr[i][1] = toMin(sc.next());
        }

        Arrays.sort(arr, (a, b) -> a[0] - b[0]);

        int freeTime = 0, prevEnd = 0;

        for (int i = 0; i < n; i++) {
            if (arr[i][0] > prevEnd)
                freeTime += arr[i][0] - prevEnd;

            prevEnd = Math.max(prevEnd, arr[i][1]);
        }

        freeTime += 1440 - prevEnd;
        System.out.println(freeTime);
    }
}

File: Daily Test/Python/Free Time - No Appointments 9.1.2026.py
def toMin(t):
    h, m = map(int, t.split(':'))
    return h*60 + m

n = int(input())
arr = []

for _ in range(n):
    s, e = input().split()
    arr.append((toMin(s), toMin(e)))

arr.sort()

freeTime = 0
prevEnd = 0

for s, e in arr:
    if s > prevEnd:
        freeTime += s - prevEnd
    prevEnd = max(prevEnd, e)

freeTime += 1440 - prevEnd
print(freeTime)